sklearn/tree/_splitter.pyx

```diff
@@ -327,10 +313,8 @@ cdef class BestSplitter(BaseDenseSplitter):
         cdef SIZE_t* constant_features = self.constant_features
         cdef SIZE_t n_features = self.n_features
 
-        cdef DTYPE_t* X = self.X
+        cdef DTYPE_t[:, :] X = self.X
```

Comment:
Are you sure this doesn't add the same kind of overhead as before? Why not just use `self.X` directly?

Reply:
Fair enough, done!